openbb_platform/CONTRIBUTING.md

```diff
@@ -1,65 +1,179 @@
-      - [Query Parameters](#query-parameters)
-      - [Data Output](#data-output)
+      - [Create Query Parameters model](#create-query-parameters-model)
+      - [Create Data Output model](#create-data-output-model)
       - [Build the Fetcher](#build-the-fetcher)
-    - [Make the new provider visible to the Platform](#make-the-new-provider-visible-to-the-platform)
-  - [Using other extension as a dependency](#using-other-extension-as-a-dependency)
-    - [Using our internal extension](#using-our-internal-extension)
-    - [Adding an external extension](#adding-an-external-extension)
-    - [The charting extension](#the-charting-extension)
-      - [Add a visualization to an existing Platform command](#add-a-visualization-to-an-existing-platform-command)
-      - [Using the `to_chart` OBBject method](#using-the-to_chart-obbject-method)
-  - [Environment and dependencies](#environment-and-dependencies)
-  - [Python package](#python-package)
-    - [Overview](#overview)
-    - [Import time](#import-time)
+    - [Make the provider visible](#make-the-provider-visible)
+  - [Manage extensions](#manage-extensions)
+    - [Add an extension as a dependency](#add-an-extension-as-a-dependency)
+  - [Manage environment and dependencies](#manage-environment-and-dependencies)
+  - [How to create a PR?](#how-to-create-a-pr)
+    - [Install pre-commit hooks](#install-pre-commit-hooks)
+    - [Branch Naming Conventions](#branch-naming-conventions)
+
+# Quick look into the OpenBB Platform
+
+The OpenBB Platform is built by the Open-Source community and is characterized by its core and extensions. The core handles data integration and standardization, while the extensions enable customization and advanced functionalities. The OpenBB Platform is designed to be used both from a Python interface and a REST API.
+
+The REST API is built on top of FastAPI and can be started by running the following command from the root:
+
+```bash
+uvicorn openbb_platform.platform.openbb_core.api.rest_api:app --host 0.0.0.0 --port 8000 --reload
+```
+
+The Python interfaces we provide to users is the `openbb` python package.
+
+The code you will find in this package is generated from a script and it is just a wrapper around the `openbb-core` and any installed extensions.
+
+When the user runs `import openbb`, `from openbb import obb` or other variants, the script that generates the package code is triggered. It detects if there are new openbb extensions installed in the environment and rebuilds the package code accordingly. If new extensions are not found, it just uses the current package version.
+
+When you are developing chances are you want to manually trigger the package rebuild.
+
+You can do that with:
+
+```python
+python -c "import openbb; openbb.build()"
+```
+
+The Python interface can be imported with:
+
+```python
+from openbb import obb
+```
+
+This document will take you through two types of contributions:
```

Comment:
We should add to this list:
- A detailed explanation of the architecture.
- A detailed explanation of the file structure.
- Nomenclature and definitions of platform-specific naming conventions.
- Docstrings.

Reply:
I think that detailed overviews are better left for other READMEs that we can link to so that this document isn't bloated with too much information from the very start. I do like the "Standards and expectations for contributions and their documentation", what do you imagine for this part?

Reply:
Some examples of contributing guidelines:

https://numpy.org/doc/stable/dev/index.html

https://github.com/streamlit/streamlit/wiki/Contributing

https://github.com/plotly/plotly.py/blob/master/contributing.md


One common theme is that environment setup and building are always at the top.

Reply:
Thanks! Yes, I moved it to the top for the OpenBB Platform section and the cookiecutter also handles that straight away.